code/modules/reagents/chemistry/holder.dm

```diff
@@ -472,10 +472,11 @@ var/const/INJECT = 5 //injection
 
 	return 0
 
-/datum/reagents/proc/clear_reagents()
+/datum/reagents/proc/clear_reagents(exclude_id)
 	for(var/reagent in reagent_list)
 		var/datum/reagent/R = reagent
-		del_reagent(R.id)
+		if(R.id != exclude_id)
+			del_reagent(R.id)
```

Comment:
This is a bad tradeoff, we're not going to add multiple checks to clear_reagents() just to fix a single issue.